### PR TITLE
Improve documentation editing environment

### PR DESCRIPTION
**Pull request by silvenon**
These changes were really connected so I found it hard to keep them in seperate commits, sorry :P

@chriseppstein told me that the documentation contributing workflow is supposed to be easy even for non-rubyists, but they were pretty messy. I'm not a rubyist so my changes didn't make the compilation process much faster, but I believe they made it easier.

I updated the gems, some of them were really old, most notably **nanoc**. After updating it I fixed some syntax deprecation warnings, updated the data sources and fixed an error I wanted to ask about. There are two lines in `lib/examples.rb` which were raising an error:

``` rb
markup_item.reps.find { |r| r.name == :default }.content_at_snapshot(:raw)
# ...
markup_item.reps.find { |r| r.name == :default }.content_at_snapshot(:raw)
```

The error said that `:raw` snapshot doesn't exist for those items. I'm not so awesome with nanoc, but changing `:raw` to `:last` seemed to fix it.

Now, for the compilation process. I removed the binstubs practice because it's kind of weird, I don't see how `./bin/something` is better from `bundle exec something`. Moreover, nanoc uses the Gemfile by default, so no `bundle exec` needed :)

Also, I made the `nanoc aco` approach secondary, as it's sluggish and ineffective (or there's something magical about it that I didin't realize). The primary approach is now `nanoc watch`+`nanoc view` (replaces `./bin/serve 3000 output`+`rake watch`), which compiles the site asynchronously (`aco` was synchronous). Those two processes are combined in a Procfile, which can be run with Foreman. Now the site can be previewed and compiled by simply running:

```
$ foreman start
```

Port 3000, same as it was before.

Lastly, I made README totally user-friendly, I hope you'll be happy with it ;)

### CODE REVIEW

**Comment by silvenon** [comment]
This hopefully fixes #990.

File: doc-src/Gemfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
 source :gemcutter
 
-gem 'nanoc', "3.2.4"
-gem 'nanoc3', "3.2.4"
-gem 'asdf'
```
Comment (scottdavis) [inline doc-src/Gemfile:5]: Curious to why we are going to a loose dependency?

Comment (silvenon) [inline doc-src/Gemfile:5]: It wasn't my intention, it's just a habit :P You can lock the version, of course.

Comment (silvenon) [inline doc-src/Gemfile:5]: While we're on this patch, I'd like to note that `asdf` was a typo, the name of the gem that does the viewing with nanoc is `adsf`.

**Comment by krisbulman** [comment]
+1 anything to make the documentation process easier, i spent hours figuring it all out, then was so exhausted with the process I never got around to contrib.

**Comment by silvenon** [comment]
Yep, me too. It's not much, but something is better than nothing :)

I figured out what that Serve part was, so I edited my first post accordingly, clarified the README and removed the Rakefile. Basically, `./bin/serve 300 output`+`rake watch` did what `foreman start` (`nanoc watch`+`nanoc view`) does now, and that is asynchronous compilation.

File: doc-src/Gemfile (hunk 1 above)
Comment (silvenon) [inline doc-src/Gemfile:5]: I locked nanoc version, I forgot I can do that myself :P

**Comment by scottdavis** [comment]
I'm really liking this except for you did it against stable and not master =(. It shouldn't be too big of a deal.

If @chriseppstein is ok ill go ahead and merge this

**Comment by silvenon** [comment]
So should I always submit pull requests against master? That part was never clear to me :) Why is stable default then?

**Comment by krisbulman** [comment]
@silvenon Pretty sure this is covered in the contrib docs http://compass-style.org/help/tutorials/contributing/#patches  
- If you are submitting features that have more than one changeset, please create a topic branch to hold the changes while they are pending merge 
- If you have single-commit patches, it is fine to keep them on master. But do keep in mind that these changesets might be cherry-picked.

**Comment by silvenon** [comment]
Oh, I should base my branches off of master then. But stable being the default results in many people submitting pull requests against stable.

**Comment by scottdavis** [comment]
I know it drives me crazy but, chris likes it that way so people can look though the stable code faster. I wish github had a way to set a default branch for people when they forked.

**Comment by silvenon** [comment]
Btw, I rebased my `doc-env` on `master` locally to see if there would be problems. It just has some minor Gemfile.lock conflicts, so it should be fine.

**Comment by silvenon** [comment]
Whoops, I realized I didn't test whether `thor` commands work properly, I just assumed they work fine. I'll let you know when I test them.

**Comment by silvenon** [comment]
Thor definitely needs some work and it's lacking instructions (current ones lead to an error). I'll do it tomorrow.

**Comment by scottdavis** [comment]
Aparently the new nanoc is suposed to not have fssm any longer but when i try this branch its exploding 

<pre>
[sdavis@Scotts-iMac:doc-src (doc-fixes)]$ foreman start
22:26:05 watch.1 | started with pid 15813
22:26:05 view.1  | started with pid 15814
22:26:06 watch.1 | Captain! We’ve been hit!
22:26:06 watch.1 | Message:
22:26:06 watch.1 | LoadError: cannot load such file -- fssm
22:26:06 watch.1 | Try installing the 'fssm' gem (`gem install fssm`) and then re-running the command.
22:26:06 watch.1 | Compilation stack:
22:26:06 watch.1 |   (empty)
22:26:06 watch.1 | Stack trace:
22:26:06 watch.1 |   0. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/nanoc-3.4.1/lib/nanoc/cli/commands/watch.rb:14:in `require'
22:26:06 watch.1 |   1. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/nanoc-3.4.1/lib/nanoc/cli/commands/watch.rb:14:in `run'
22:26:06 watch.1 |   2. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/nanoc-3.4.1/lib/nanoc/cli/command_runner.rb:14:in `block in call'
22:26:06 watch.1 |   3. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/nanoc-3.4.1/lib/nanoc/cli/error_handler.rb:63:in `handle_while'
22:26:06 watch.1 |   4. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/nanoc-3.4.1/lib/nanoc/cli/error_handler.rb:26:in `handle_while'
22:26:06 watch.1 |   5. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/nanoc-3.4.1/lib/nanoc/cli/command_runner.rb:13:in `call'
22:26:06 watch.1 |   6. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/cri-2.3.0/lib/cri/command_dsl.rb:185:in `block in runner'
22:26:06 watch.1 |   7. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/cri-2.3.0/lib/cri/command.rb:296:in `call'
22:26:06 watch.1 |   8. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/cri-2.3.0/lib/cri/command.rb:296:in `run_this'
22:26:06 watch.1 |   9. /Users/sdavis/.rvm/gems/ruby-1.9.3-p0-perf/gems/cri-2.3.0/lib/cri/command.rb:249:in `run'
22:26:06 watch.1 | If you believe this is a bug in nanoc, please do report it at
22:26:06 watch.1 | -> https://github.com/ddfreyne/nanoc/issues/new <-
22:26:06 watch.1 | A detailed crash log has been written to ./crash.log.
22:26:06 watch.1 |   ... 8 more lines omitted. See full crash log for details.
22:26:06 watch.1 | exited with code 1
22:26:06 system  | sending SIGTERM to all processes
SIGTERM received
22:26:06 view.1  | exited with code 0
</pre>

**Comment by scottdavis** [comment]
nvm got it

**Comment by scottdavis** [comment]
So I'm tracking this in a branch now here https://github.com/chriseppstein/compass/tree/doc-fixes with some changes i made since I've removed blue print and merged with master. going to let it live here for a few days until we can get a docs site up for blueprint.

**Comment by scottdavis** [comment]
Nevermind chris talked me into merging it .. its all in master now

**Comment by silvenon** [comment]
Noooooo, I was just about to add some hardcore Thor stuff :P I'll open another pull request then, on master this time.

**Comment by scottdavis** [comment]
That's fine it's got I can merge it

**Pull request closed without merging**